x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/Querier.java

```diff
@@ -91,24 +88,19 @@
 public class Querier {
     private static final Logger log = LogManager.getLogger(Querier.class);
 
-    private final PlanExecutor planExecutor;
+    private final NamedWriteableRegistry writeableRegistry;
     private final SqlConfiguration cfg;
-    private final int size;
     private final Client client;
-    @Nullable
-    private final QueryBuilder filter;
-
-    public Querier(SqlSession sqlSession) {
-        this.planExecutor = sqlSession.planExecutor();
-        this.client = sqlSession.client();
-        this.cfg = sqlSession.configuration();
-        this.filter = cfg.filter();
-        this.size = cfg.pageSize();
+
+    public Querier(Client client, NamedWriteableRegistry writeableRegistry, SqlConfiguration cfg) {
```

Comment:
Also making the dependencies of `Querier` more precise and removing the redundant `size` and `filter` fields.

Reply:
Not sure if this is an improvement since all these arguments are available on the session - it's now on the caller to unpack these items.
It's unclear to me what the gain is.

Reply:
`SqlSession` is a very powerful class, so currently we're exposing a lot of power to `Querier` by default. Also, it requires callers to actually have an `SqlSession` which is asking for a lot. Since there is only one caller using `Querier` plus the `QuerierTests` that had to mock `SqlSession` the amount of code is even reduced.

Not feeling super strong about this change but I think it helps to avoid errors like the one fixed by this PR.

Reply:
> I think it helps to avoid errors like the one fixed by this PR.

Wondering if this still holds with current proposed change.
(I personally tend to favor shorter args lists too, but feel as weak about it in this case.)

Reply:
I still think that it holds but I'm reverting the change because there seems to be a general preference of the `SqlSession` param.